src/util-decode-mime.c

```diff
@@ -1912,6 +1946,11 @@ static int ProcessMimeHeaders(const uint8_t *buf, uint32_t len,
                 }
                 memcpy(entity->filename, bptr, blen);
                 entity->filename_len = blen;
+
+                if (truncated_name) {
+                    state->stack->top->data->anomaly_flags |= ANOM_LONG_FILENAME;
```

Comment:
why is the flag set in 2 places here but not at the block around line 1999 below?

Reply:
Thank; missed that.

I'm not sure why the anomaly flags are set off of `msg` *and* `top`. That seems odd.